### PR TITLE
Linkdrop e2e tests

### PR DESCRIPTION
**Pull request by esaminu**
Added e2e tests for linkdrop flows `/linkdrop`:
* Claim linkdrop to logged in account using dropdown
* Create new account and claim linkdrop to that account 
* Login to new account with seedphrase and claim to that account

### CODE REVIEW

**Review by MaximusHaximus — CHANGES_REQUESTED** [review]
Just a few things in-line -- nice work, we're really getting into the nitty gritty parts of the wallet here :) 💖

File: packages/e2e-tests/linkDrop/signedInUser.spec.js [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+    let linkdropSenderAccount,
+        linkdropReceiverAccount,
+        linkdropContractAccount,
+        linkdropKeyPair,
+        linkdropTransferNEARAmount = "2.5";
+
+    const LINKDROP_ACCESS_KEY_ALLOWANCE = new BN(parseNearAmount("1.0"));
+    const linkdropClaimableAmount = new BN(parseNearAmount(linkdropTransferNEARAmount)).sub(
+        LINKDROP_ACCESS_KEY_ALLOWANCE
+    );
```
[the remark this answers is not in the document]
Comment (esaminu) [reply packages/e2e-tests/linkDrop/signedInUser.spec.js]: Single lined for now. Made formatter modifications to allow slightly longer lines 👍

File: packages/e2e-tests/linkDrop/signedInUser.spec.js [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+        });
+        linkdropKeyPair = KeyPairEd25519.fromRandom();
+        // send linkdropTransferNEARAmount Ⓝ to contract
+        await linkdropSenderAccount.account.functionCall(
+            linkdropContractAccount.account.accountId,
+            "send",
+            { public_key: linkdropKeyPair.publicKey.toString() },
+            null,
+            new BN(parseNearAmount(linkdropTransferNEARAmount))
+        );
```
[the remark this answers is not in the document]
Comment (esaminu) [reply packages/e2e-tests/linkDrop/signedInUser.spec.js]: Done 👍

File: packages/e2e-tests/linkDrop/newUser.spec.js [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+            createRandomBankSubAccount(),
+        ]).catch((e) => {
+            throw new Error(
+                "Cannot run test suite, linkdrop sender, receiver and contract accounts not successfully created",
+                {
+                    cause: e,
+                }
+            );
+        });
+        linkdropKeyPair = KeyPairEd25519.fromRandom();
```
[the remark this answers is not in the document]
Comment (esaminu) [reply packages/e2e-tests/linkDrop/newUser.spec.js]: Receiving this from new setup function now 👍 . It's the keypair for the first linkdrop we create

File: packages/e2e-tests/linkDrop/newUser.spec.js [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+        await expect(new BN(parseNearAmount(nearBalance)).gte(linkdropClaimableAmount)).toBe(true);
+    });
+    test("claims linkdrop to new account", async ({ page, context }) => {
+        await context.grantPermissions(["clipboard-read", "clipboard-write"]).catch(test.skip);
+        // skip test on mainnet
+        if (walletNetwork === WALLET_NETWORK.mainnet) {
+            test.skip();
+        }
+
+        linkdropKeyPair = KeyPairEd25519.fromRandom();
```
[the remark this answers is not in the document]
Comment (esaminu) [reply packages/e2e-tests/linkDrop/newUser.spec.js]: Changed to a constant 👍 . It's the sender keypair for the second linkdrop we create

**Comment by stefanopepe** [comment]
@esaminu where do you suggest adding comments to help LinkDrop developers?
I see three files, one is the model, and then two separate tests for existing accounts and new accounts.

See an example of the result from here: https://github.com/bitcoin/bitcoin/blob/master/src/test/transaction_tests.cpp

**Comment by esaminu** [comment]
@stefanopepe I added some basic integration docs [here](https://github.com/near/near-wallet/blob/4723266cabe5443b6c6bb043a72915f33373f38c/packages/frontend/docs/Linkdrop.md). I want to add a table of contents on the [root README file](https://github.com/near/near-wallet/blob/4723266cabe5443b6c6bb043a72915f33373f38c/packages/frontend/docs/Linkdrop.md) for all flows that links to test code and docs like above (if available). For inline comments, I suggest we add it to [`LinkdropAccountManger`](https://github.com/near/near-wallet/blob/4723266cabe5443b6c6bb043a72915f33373f38c/packages/e2e-tests/utils/account.js#L76) (once approved) and some on the steps on [test spec for new users](https://github.com/near/near-wallet/blob/4723266cabe5443b6c6bb043a72915f33373f38c/packages/e2e-tests/linkDrop/newUser.spec.js) since a lot of the steps are quite redundant in the spec and `LinkdropAccountManager` is responsible for account creation, contract deployment and sending the funds to be dropped.

**Review by stefanopepe — APPROVED** [review]
The documentation page looks good to me! 
@MaximusHaximus the rest seems unchanged from the previous review request, so it's gtg for me

File: packages/e2e-tests/utils/account.js [hunk 5]
```diff
@@ -4,10 +4,14 @@ const E2eTestAccount = require("./E2eTestAccount");
 const getBankAccount = async () => {
     const { BANK_ACCOUNT: accountId, BANK_SEED_PHRASE: seedPhrase } = process.env;
     const account = new E2eTestAccount(accountId, seedPhrase, { accountId: nearApiJsConnection.config.networkId });
-    await account.initialize();
-    return account;
+    return account.initialize();
 };
 
+function generateTestAccountId() {
+    return `test-playwright-account-${Date.now()}-${Math.floor(Math.random() * 1000) % 1000}`;
```
Comment (MaximusHaximus) [inline packages/e2e-tests/utils/account.js:11]: Did we decide to abbreviate this? :D

**Review by MaximusHaximus — APPROVED** [review]
LGTM, just a couple of minor things in-line that aren't blockers.  If you've got nothing else pending change-wise, feel free to go ahead and merge this sucker :)

**Comment by MaximusHaximus** [comment]
@esaminu Is this PR OK to land or is there something blocking it?

File: packages/e2e-tests/utils/account.js (hunk 5 above)
Comment (esaminu) [reply packages/e2e-tests/utils/account.js:11]: Done here: https://github.com/near/near-wallet/pull/2174/files#diff-c2c58188c2514e87ba540c8f2d7d767e40fe9f361f3f6bae1ab31e68f75c56cbR20

File: packages/frontend/docs/Linkdrop.md [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+
+* [What is a linkdrop](#what-is-a-linkdrop)
+* [Creating a linkdrop](#creating-a-linkdrop)
+* [Wallet linkdrop page](#wallet-linkdrop-page)
+
+## What is a linkdrop?
+
+Linkdrops allow you to send funds to anyone via a link. The reciever of a linkdrop can add the funds to their existing account or create a new account to claim the funds.
+
+![](https://i.imgur.com/P5WMg1i.png)
```
[the remark this answers is not in the document]
Comment (esaminu) [reply packages/frontend/docs/Linkdrop.md]: Done 👍

File: packages/frontend/docs/Linkdrop.md [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+
+
+## Creating a linkdrop
```
[the remark this answers is not in the document]
Comment (esaminu) [reply packages/frontend/docs/Linkdrop.md:14]: Done 👍 . Code example uses `near` TLA as well.

**Comment by esaminu** [comment]
> @esaminu Is this PR OK to land or is there something blocking it?

@MaximusHaximus just the requested changes to doc. Should be good to go now! 👍

**Pull request merged**